### PR TITLE
Create separate pull requests for each post and check existing posts based on the title

### PR DESCRIPTION
**Pull request by arash77**
This PR creates separate pull requests for each post and checks if there are any existing posts with the same title that contains the url.

### CODE REVIEW

File: app/feed_bot.py [hunk 1]
```diff
@@ -49,16 +48,17 @@ def main():
...
-        f"Update from feeds input bot since {utils_obj.start_date.strftime('%Y-%m-%d')}"
-    )
-    feeds_processed_str = "- " + "\n- ".join(feeds_processed)
-    body = f"This PR created automatically by feed bot.\n\nFeeds processed:\n{feeds_processed_str}"
-    utils_obj.create_pull_request(title, body)
+            utils_obj.process_entry(entry_data)
+            title = f"Update from feeds: {entry.link}"
+            feeds_processed_str = f"[{entry.title}]({entry.link})"
+            body = (
+                f"This PR created automatically by feed bot.\n"
```
Comment (bgruening) [inline app/feed_bot.py:57]:
```suggestion
                f"This PR is created automatically by a feed bot.\n"
```

File: app/utils.py [hunk 2]
```diff
@@ -41,18 +41,11 @@ def __init__(self, bot_path, item_type):
...
-            if pr_file.filename.startswith(self.bot_path)
-        )
-        git_tree = self.repo.get_git_tree(self.repo.default_branch, recursive=True)
-        self.existing_files.update(
-            file.path
-            for file in git_tree.tree
-            if file.path.startswith(self.bot_path) and file.path.endswith(".md")
-        )
+        self.existing_files = set()
+        for pr in self.repo.get_pulls(state=["open", "closed"]):
```
Comment (bgruening) [inline app/utils.py:45]: you can filter here more: https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-pull-requests

e.g. for head or maybe base?

Comment (arash77) [reply app/utils.py]: https://github.com/usegalaxy-eu/galaxy-social-assistent/pull/3/commits/98266ec86953d89384ea80f6d91e83301e773f83

File: app/utils.py [hunk 3]
```diff
@@ -17,49 +18,39 @@ def __init__(self, bot_path, item_type):
...
-            if file.path.startswith(self.bot_path) and file.path.endswith(".md")
-        )
-        self.branch_name = (
-            f"{self.bot_path}-update-{datetime.now().strftime('%Y%m%d%H%M%S')}"
-        )
-        self.repo.create_git_ref(
-            ref=f"refs/heads/{self.branch_name}",
-            sha=self.repo.get_branch("main").commit.sha,
-        )
+        self.existing_files = {
```
Comment (bgruening) [inline app/utils.py:47]: I don't find this very readable. Why not pull the for-loop out and construct a set inside the for loop?

File: app/utils.py [hunk 4]
```diff
@@ -17,49 +18,39 @@ def __init__(self, bot_path, item_type):
...
-                raise ValueError(
-                    f"No media found in the config file for {list_items} in {item_type}"
-                )
-            elif list_items.get("format") is None:
-                raise ValueError(
-                    f"No format found in the config file for {list_items} in {item_type}"
-                )
+            raise ValueError(f"No {self.item_type} found in the config file")
+
+        errors = [
```
Comment (bgruening) [inline app/utils.py:30]: Can you enhance readability here? You can pull out the for loop and just keep adding a error-list imho.

**Pull request merged**